osquery/tables/networking/darwin/wifi.mm

```diff
@@ -25,25 +25,52 @@
     "/Library/Preferences/SystemConfiguration/"
     "com.apple.airport.preferences.plist";
 
-const std::map<std::string, std::string> kKnownWifiNetworkKeys = {
+// In 10.14 and prior, there was an "auto_login" key.
+const std::map<std::string, std::string> kKnownWifiNetworkKeysPreCatalina = {
+    {"auto_login", "AutoLogin"}, {"last_connected", "LastConnected"}};
+
+const std::map<std::string, std::string> kKnownWifiNetworkKeysCommon = {
     {"ssid", "SSID"},
     {"network_name", "SSIDString"},
     {"security_type", "SecurityType"},
     {"roaming_profile", "RoamingProfileType"},
-    {"auto_login", "AutoLogin"},
-    {"last_connected", "LastConnected"},
     {"captive_portal", "Captive"},
     {"roaming", "SPRoaming"},
     {"passpoint", "Passpoint"},
     {"possibly_hidden", "PossiblyHiddenNetwork"},
     {"disabled", "Disabled"},
     {"temporarily_disabled", "TemporarilyDisabled"}};
 
-// Check if we are running on OS X 10.9, where the key in plist is different
+// The name of the "last_connected" key changed in 10.15.
+const std::map<std::string, std::string> kKnownWifiNetworkKeysPostCatalina = {
+    {"last_connected", "LastAutoJoinAt"}};
+
+// Adjust the keys to read, based on the version of macOS
+Status getKnownWifiNetworkKeys(std::map<std::string, std::string>& keys) {
+  auto qd = SQL::selectAllFrom("os_version");
```

Comment:
I don't like this approach but I see it used in several other places so we can follow up and fix this later.

Reply:
It's a table with one row in this case, but I can see it has been a performance problem in other cases in the past?